src/Symfony/Component/Console/Command/LockableTrait.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Command;
+
+use Symfony\Component\Console\Exception\RuntimeException;
+use Symfony\Component\Filesystem\LockHandler;
+
+/**
+ * Basic lock feature for commands.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+trait LockableTrait
+{
+    protected $lock;
+
+    /**
+     * Locks a command.
+     *
+     * @param string $name
+     * @param bool   $blocking
+     *
+     * @return bool
+     *
+     * @throws RuntimeException
+     */
+    protected function lock($name, $blocking = false)
```

Comment:
What about making the `name` able to accept full paths as well? We usually put our locks into `%projectdir%/var/run` so doing something like

```php
$lockPath = null;
$lockName = $name ?: $this->getName();

if ($name && $fs->isAbsolutePath($name)) {
    $lockPath = dirname($name);
    $lockName = basename($name);
}
```
would be great. Or is that look too much?